src/dht.cpp

```diff
@@ -1048,40 +1059,31 @@ Dht::searchStep(Search& sr)
                     return sn.candidate or sn.node->isExpired(now);
                 }) == sr.nodes.size())
         {
-            unsigned added = 0;
-            if (not sr.refilled) {
-                added = sr.refill(sr.af == AF_INET ? buckets : buckets6, now);
-                sr.refilled = true;
+            DHT_ERROR("[search %s IPv%c] expired", sr.id.toString().c_str(), sr.af == AF_INET ? '4' : '6');
+            // no nodes or all expired nodes
+            sr.expired = true;
+            // reset refilled since the search is now expired.
+            sr.refilled = false;
```

Comment:
Indeed. I think we can even get rid of `Search::refilled` member. However, there should be some delay between two refills in order to avoid traffic catastrophies.